iFixit/react-commerce
# Vulcan: Problem: TOC: Display like mobile causes list

## Issue
The desktop-width TOC on problem pages is very convenient, but the lack of a "Causes" header and the icons shown on the mobile causes list give the page more of a "generic wiki feel". The problem page mockup has already been updated to reflect how we'd like the TOC to look: 

![Image](https://github.com/iFixit/ifixit/assets/89873512/0e999900-582a-4ce9-b89a-a8bfa94ab6ae)

[Mock-up](https://www.figma.com/file/QVLUEpDVoyiMM4MXEGq1L9/iFixit---Problems?type=design&node-id=296-44379&mode=design&t=AWzeP1tPVtFUcgqV-4)

_This one had a few moving parts. If I can help clarify anything, don't hesitate to reach out._

## CR/QA

`Troubleshooting/Refrigerator/Ice+Maker+Not+Making+Ice/481782` is a solid test-case with many causes and related problems.

<details>
<summary>Post-fix Video</summary>

https://github.com/iFixit/react-commerce/assets/1634505/90a123fc-d1f2-4df2-98f5-5070a77e7c6b

</details>

We want to make sure:

- Adds a "Causes" heading
- Adds headings to each TOC section (e.g. "Causes", etc)
- Adds icons in front of each section
  - Custom icon for "Introduction" and "Conclusions"
  - Numbers for solutions
  - Don't display "Related Problems" in the TOC _at all_ (not at any breakpoint, mobile or desktop)
    - We _do_ still want to show the Related Problems item in the causes list at the top of the page at mobile widths

Later additions:

- Add additional headings to TOC; "Troubleshooting" and "Conclusions"
- Make "Related Problems" sidebar sticky and scrollable to match TOC UX

Closes https://github.com/iFixit/ifixit/issues/50725
